.github/workflows/ci.yml

```
@@ -50,8 +57,139 @@ jobs:
+  diagrams:
+    name: Mermaid Diagrams
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Node and mermaid-cli
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y nodejs npm
+          sudo npm i -g @mermaid-js/mermaid-cli
+      - name: Compile .mmd to .svg
+        run: |
```

actionlint (1.7.8)

shellcheck reported issue in this script: SC2044:warning:3:10: For loops over find output are fragile. Use find -exec or a while read loop

(shellcheck)